docs/choreolib/auto-factory.md

```diff
@@ -1,109 +1,268 @@
+
+Below is an example auto routine from the 2024 season, taking advantage of the `AutoRoutine` paradigm:
+
+!!! note
+    Be careful when using `Commands.either()` to achieve branching behavior. The resulting composition will require subsystems from both the `onTrue` and `onFalse` commands, which may block default commands that would otherwise be expected to run. As shown in the example below, chaining triggers is the recommended pattern for creating a branching auto.
+
+```java
+public AutoRoutine branching2024Auto() {
+    AutoRoutine routine = autoFactory.newRoutine("branching2024Auto");
+
+    // This routine uses segments between pre-defined handoff points.
+    // Expand the tooltip for information about their naming convention -> (1)
+
+    // Load the routine's trajectories
+    AutoTrajectory startToC2 = routine.trajectory("startToC2");
+    AutoTrajectory C2toM1 = routine.trajectory("C2toM1");
+    AutoTrajectory M1toM2 = routine.trajectory("M1toM2");
+    AutoTrajectory M2toM3 = routine.trajectory("M2toM3");
+    AutoTrajectory M1toScore = routine.trajectory("M1toScore");
+    AutoTrajectory M2toScore = routine.trajectory("M2toScore");
+    AutoTrajectory M3toScore = routine.trajectory("M3toScore");
+    AutoTrajectory scoreToM2 = routine.trajectory("scoreToM2");
+    AutoTrajectory scoreToM3 = routine.trajectory("scoreToM3");
+
+    // When the routine starts, reset odometry, shoot the first gamepiece, then go to the "C2" location
+    routine.active().onTrue(
+        Commands.sequence(
+            routine.resetOdometry(startToC2),
+            shooterSubsystem.shoot(),
+            startToC2.cmd()
+        )
+    );
+
+    // Pick up and shoot the gamepiece at the "C2" location, then go to the "M1" location
+    startToC2.active().whileTrue(intakeSubsystem.intake());
+    startToC2.done().onTrue(shooterSubsystem.shoot().andThen(C2toM1.cmd()));
+
+    // Run the intake when we are approaching a gamepiece
+    C2toM1.active()
+        .or(scoreToM2.active())
+        .or(scoreToM3.active())
+        .or(M1toM2.active())
+        .or(M2toM3.active())
+        .whileTrue(intakeSubsystem.intake());
+
+    // If we picked up the gamepiece, go score, then go to the next midline location
+    // If we didn't pick up the gamepiece, go directly to the next midline location
 
-  public void autonomousInit() {
-    autoChooser.selectedCommand().schedule();
-  }
+    // M1
+    Trigger atM1 = C2toM1.done();
```

Comment:
The behavior that AutoTrajectory.done() is only true for one cycle isn't mentioned anywhere on this page and is key to this example working as desired. This behavior should be explicitly mentioned on this page, as it's critical for users to know.